### PR TITLE
Judge before UseCachedDecoderOut

### PR DESCRIPTION
**Pull request by HieDean**
When the cached decoder out of the entire batch exists, then use it.

### CODE REVIEW

File: sherpa-onnx/csrc/online-transducer-greedy-search-decoder.cc [hunk 1]
```diff
@@ -89,9 +89,23 @@ void OnlineTransducerGreedySearchDecoder::Decode(
...
+  for (const auto &r : *result) {
+    if (!r.decoder_out) {
+      is_batch_decoder_out_cached = false;
+      break;
+    }
+  }
+  if (is_batch_decoder_out_cached) {
+    auto &r = result->front();
+    std::vector<int64_t> decoder_out_shape = r.decoder_out.GetTensorTypeAndShapeInfo().GetShape();
+    decoder_out = Ort::Value::CreateTensor<float>(model_->Allocator(), decoder_out_shape.data(), decoder_out_shape.size());
```
Comment (csukuangfj) [inline sherpa-onnx/csrc/online-transducer-greedy-search-decoder.cc:104]: This line only allocates memory for decoder_out but it does not initialize it.

Please use
https://github.com/k2-fsa/sherpa-onnx/blob/eeda1e190e7fb9132442190aa3e10699b486c00e/sherpa-onnx/csrc/cat-test.cc#L126
to concatenate the `r.decoder_out` along the batch dimension.

**Comment by csukuangfj** [comment]
By the way, could you follow
https://github.com/k2-fsa/sherpa-onnx/blob/eeda1e190e7fb9132442190aa3e10699b486c00e/sherpa-onnx/csrc/online-wenet-ctc-model.cc#L74-L75
to replace `Clone()` with `View()` in other code in a separate PR?

File: sherpa-onnx/csrc/online-transducer-greedy-search-decoder.cc (hunk 1 above)
Comment (HieDean) [reply sherpa-onnx/csrc/online-transducer-greedy-search-decoder.cc:104]: Sorry for my mistake. `Cat()` seems to be a more overloaded option.

**Comment by csukuangfj** [comment]
Thank you for your first contribution!

**Pull request merged**